server/data/athenaClient.test.ts

```diff
@@ -0,0 +1,178 @@
+import type { GetQueryExecutionOutput, GetQueryResultsOutput, StartQueryExecutionOutput } from '@aws-sdk/client-athena'
+import { QueryExecutionState as ExecutionState } from '@aws-sdk/client-athena'
+
+import AthenaClient, { AthenaConfig } from './athenaClient'
+import { S3BucketConfig } from './s3Client'
+
+const bucketConfig = { bucket: 'bucket1' } as S3BucketConfig
+const athenaConfig = {
+  workGroup: 'work-group-1',
+  database: 'database_1',
+} as AthenaConfig
+
+const athena = {
+  send: jest.fn().mockReturnThis(),
+}
+
+jest.mock('@aws-sdk/client-athena', () => {
+  const { QueryExecutionState, GetQueryExecutionCommand, GetQueryResultsCommand, StartQueryExecutionCommand } =
+    jest.requireActual('@aws-sdk/client-athena')
+  return {
+    AthenaClient: jest.fn(() => athena),
+    QueryExecutionState,
+    GetQueryExecutionCommand,
+    GetQueryResultsCommand,
+    StartQueryExecutionCommand,
+  }
+})
+
+describe('AthenaClient', () => {
+  let athenaClient: AthenaClient
+
+  const fakeQueryId = '11111111-2222-3333-4444-555555555555'
+
+  function fakeStartQueryExecution(): StartQueryExecutionOutput {
+    return {
+      QueryExecutionId: fakeQueryId,
+    }
+  }
+
+  function fakeQueryExecutionStatus(status: ExecutionState): GetQueryExecutionOutput {
+    return {
+      QueryExecution: {
+        QueryExecutionId: fakeQueryId,
+        WorkGroup: athenaConfig.workGroup,
+        Status: { State: status },
+      },
+    }
+  }
+
+  beforeEach(() => {
+    athenaClient = new AthenaClient(athenaConfig)
+  })
+
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
+  describe('startExecution()', () => {
+    it('passes parameters to Athena', async () => {
+      const awsResponse = fakeStartQueryExecution()
+      athena.send.mockResolvedValue(awsResponse)
+      const sqlQuery = 'SELECT * FROM prison_population;'
+      const response = await athenaClient.startExecution(sqlQuery)
+      expect(response).toEqual(fakeQueryId)
+      // expect the first parameter of the first call to include work group and database
+      const [command] = athena.send.mock.calls[0]
+      const { QueryString, WorkGroup, QueryExecutionContext } = command.input
+      const { Database } = QueryExecutionContext
+      expect(QueryString).toEqual(sqlQuery)
+      expect(WorkGroup).toEqual(athenaConfig.workGroup)
+      expect(Database).toEqual(athenaConfig.database)
+    })
+  })
+
+  describe('getExecutionStatus()', () => {
+    it('returns execution status', async () => {
+      const awsResponse = fakeQueryExecutionStatus(ExecutionState.SUCCEEDED)
+      athena.send.mockResolvedValue(awsResponse)
+      const response = await athenaClient.getExecutionStatus(fakeQueryId)
+      expect(response).toEqual(ExecutionState.SUCCEEDED)
+    })
+  })
+
+  describe('executionCompletion()', () => {
+    function makeTest(status: ExecutionState) {
```

Comment:
This makes sense to me and it's clean but just FYI Jest has few ways of doing table testing, e.g. [`describe.each(table)()`](https://jestjs.io/docs/api#describeeachtablename-fn-timeout).

But I haven't played with it yet, it may be worse in this case? Just wanted to let you know in case you missed this trick.

Reply:
oh i like that